npc/selector.go

```diff
@@ -12,25 +12,39 @@ import (
 )
 
 type selectorSpec struct {
-	key         string          // string representation (for hash keying/equality comparison)
-	selector    labels.Selector // k8s Selector object (for matching)
-	policyTypes []policyType    // If non-empty, then selectorSpec is a target selector for given policyTypes.
+	key               string          // string representation (for hash keying/equality comparison)
+	podSelector       labels.Selector // k8s Selector object (for matching pods)
+	namespaceSelector labels.Selector // k8s Selector object (for matching namespaces)
```

Comment:
What is the relationship between `namespaceSelector` and `nsName` ?
I'm wondering if `nsName` can be replaced by a selector matching that one namespace.

Reply:
nsName is just used to generate appropriate comments for the iptable rules. No relationship between namespaceSelector and nsName

Reply:
For the record, the key point I was missing is that "a selector matching that one namespace" is not something you can expect to construct, given an arbitrary namespace.  Selectors can only match on labels, and the namespace may not have any labels.